### ISSUE TITLE
`temp`/`let` lose `is default` of hashes and arrays

### ISSUE THREAD

**Issue by zoffixznet** [rakudo/rakudo#1432]
```
<Zoffix__> m: (my @a is default(Nil))[3] = 42; say @a[1]:exists; say @a; { let @a; Nil }; say @a[1]:exists; say @a
<camelia> rakudo-moar 0a1641e94: OUTPUT: «False␤[Nil Nil Nil 42]␤True␤[(Any) (Any) (Any) 42]␤»

<Zoffix__> m: (my @a is default(Nil))[3] = 42; say @a[1]:exists; say @a; { temp @a; Nil }; say @a[1]:exists; say @a
<camelia> rakudo-moar 0a1641e94: OUTPUT: «False␤[Nil Nil Nil 42]␤True␤[(Any) (Any) (Any) 42]␤»

Zoffix__> m: (my %h is default(Nil))<a> = Nil; say %h; { temp %h; Nil }; say %h
<camelia> rakudo-moar 0a1641e94: OUTPUT: «{a => Nil}␤{a => (Any)}␤»
```

**Comment by zoffixznet** [comment]
More accurately: any existing `Nil`s in the original get converted to an `Any` because the intermediary container doesn't have a descriptor.

**Comment by zoffixznet** [comment]
This is now fixed for hashes and will automatically get fixed for arrays when #1435 is fixed. [Tests committed](https://github.com/perl6/roast/commit/79c8fb720478948985328469f3d7c91cc2bb6deb) with Array tests todo-fudged.
